### PR TITLE
Update frontend UI for changing place on place page

### PR DESCRIPTION
**Pull request by juliawu**
Update the styling of the search bar used to change places on the place explorer to match figma mocks. This improves the current UI by preventing two search bars from showing up next to each other on page load.

Screenshots:
Screenshot 2023-12-21 at 12 56 06 PM
Screenshot 2023-12-21 at 12 55 56 PM
Screenshot 2023-12-21 at 12 55 33 PM

Screenshot 2023-12-21 at 12 44 33 PM
Screenshot 2023-12-21 at 12 44 44 PM
Screenshot 2023-12-21 at 12 44 59 PM

### CODE REVIEW

File: static/css/place/change_place_search_bar.scss [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  align-items: flex-end;
+  display: flex;
+  height: 100%;
+  width: 100%;
+}
+
+#change-place {
+  display: flex;
+  height: fit-content;
+  justify-content: right;
```
Comment (dwnoble) [inline static/css/place/change_place_search_bar.scss:39]: Can you try left-aligning when in mobile? The right alignment doesn't look quite right to me

**Review by dwnoble — CHANGES_REQUESTED** [review]

File: static/css/place/change_place_search_bar.scss [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Styles for "Change Place" search bar on place pages */
+
+// "Change Place" search bar colors
+$GM3-sys-light-primary: #0b57d0;
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/css/place/change_place_search_bar.scss]: Good call out, migrated to `_colors.scss`.

File: static/css/place/change_place_search_bar.scss [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+#place-list {
+  align-items: center;
+  display: flex;
+  flex-shrink: 0;
+  flex-wrap: wrap;
+  margin-left: 32px;
+}
+
+#close-icon {
+  background: url("../images/icon-cancel-24px.svg") no-repeat;
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/css/place/change_place_search_bar.scss]: Swaped to using Materias Icons Outlined, using the default weight but keeping the same color used in the original svg from the mocks.

File: static/js/place/place_search.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        {toggleText}
+      </div>
+      <div id="search" style={{ display: `${showSearchBar ? "" : "none"}` }}>
+        <div id="search-text">
+          <div id="search-instructions">Enter a place name</div>
+          <div id="location-field">
+            <input id="place-autocomplete" placeholder="" type="text" />
+          </div>
+        </div>
+        <div id="close-icon" onClick={() => setShowSearchBar(false)}></div>
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/place/place_search.tsx]: Thanks, didn't know about this. Swapped to className everywhere, but kept a few ids that are used in the webdriver tests.

File: static/css/place/change_place_search_bar.scss (hunk 1 above)
Comment (juliawu) [reply static/css/place/change_place_search_bar.scss]: Tried many CSS/flexbox variations to get left-aligning on wrap, ended up setting left-align behavior using a `@media` rule with a breakpoint found empirically. PTAL and let me know if you dislike this approach.

File: static/images/icon-cancel-24px.svg
Comment (dwnoble) [inline static/images/icon-cancel-24px.svg:1]: Remove this file?

**Review by dwnoble — APPROVED** [review]
Nice! Just a few nits.

File: static/images/icon-cancel-24px.svg
Comment (juliawu) [reply static/images/icon-cancel-24px.svg:1]: Thanks for the catch! Removed.

File: static/css/place/change_place_search_bar.scss [hunk 5]
```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright 2023 Google LLC
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/css/place/change_place_search_bar.scss]: Updated!

**Pull request merged**